odpf/siren/v1/siren.proto

```diff
@@ -0,0 +1,343 @@
+    string workspace = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message ExchangeCodeResponse {
+    bool ok = 1;
+}
+
+message GetAlertCredentialsRequest{
+    string team_name = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message Critical {
+    string channel = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message Warning {
+    string channel = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+}
+
+message SlackConfig {
+    Critical critical = 1 [(validate.rules).message.required = true];
+    Warning warning = 2 [(validate.rules).message.required = true];
+}
+
+message GetAlertCredentialsResponse {
+    string entity = 1;
+    string team_name = 2;
+    string pagerduty_credentials = 3;
+    SlackConfig slack_config = 4;
+}
+
+message UpdateAlertCredentialsRequest {
+    string entity = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string team_name = 2;
+    string pagerduty_credentials = 3 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    SlackConfig slack_config = 4 [(validate.rules).message.required = true];
+}
+message UpdateAlertCredentialsResponse{}
+
+
+message SendSlackNotificationRequest{
+    string provider = 1;
+    string entity = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string message = 3;
+    string receiver_name = 4 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string receiver_type = 5 [(validate.rules).string = {in: ["channel", "user"]}];;
+    repeated google.protobuf.Struct blocks = 6;
+}
+
+message SendSlackNotificationResponse {
+    bool ok = 1;
+}
+
+message ListRulesRequest {
+    string namespace = 1;
+    string entity= 2;
+    string group_name = 3;
+    string status = 4;
+    string template = 5;
+}
+
+message Variables {
+    string name = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string type = 2;
+    string value = 3 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string description = 4;
+}
+
+message Rule {
+    uint64 id = 1;
+    string name = 2;
+    string entity = 3;
+    string namespace = 4;
+    string group_name = 5;
+    string template = 6;
+    string status = 7;
+    google.protobuf.Timestamp created_at = 8;
+    google.protobuf.Timestamp updated_at = 9;
+    repeated Variables variables = 10;
+}
+
+message ListRulesResponse {
+    repeated Rule rules = 1;
+}
+
+message UpdateRuleResponse {
+    Rule rule = 1;
+}
+
+message UpdateRuleRequest {
+    uint64 id = 1;
+    string name = 2;
+    string entity = 3 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string namespace = 4 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string group_name = 5 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string template = 6 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
+    string status = 7 [(validate.rules).string = {in: ["enabled", "disabled"]}];
```

Comment:
Can we can convert status field to enum?

Reply:
Not doing this because downstream service accepts `strings` in order to implement this we need to put unnecessary type conversion in the middle which I would like to avoid.